Docs: Add SameSite troubleshooting to SAML docs (#75590)
grafana/grafana: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md

Following are common issues found in configuring SAML authentication in Grafana and how to resolve them.

### Infinite redirect loop / User gets redirected to the login page after successful login on the IdP side

> If you are experiencing an infinite redirect loop (when `auto_login = true`) or being redirected to the login page after successful login, it is likely that the `grafana_session` cookie's SameSite setting is set to `Strict` which means that the `grafana_session` cookie is not sent to Grafana in case of cross-site requests. To resolve this issue, set the `security.cookie_samesite` option to `Lax` in the Grafana configuration file.

Reviewer: ```suggestion
If you experience an infinite redirect loop when `auto_login = true` or redirected to the login page after successful login, it is likely that the `grafana_session` cookie's SameSite setting is set to `Strict`. This setting prevents the `grafana_session` cookie from being sent to Grafana during cross-site requests. To resolve this issue, set the `security.cookie_samesite` option to `Lax` in the Grafana configuration file.
```
